src/pymmcore_widgets/_hcs_widget/_calibration_widget.py

```diff
@@ -0,0 +1,597 @@
+                well_list.extend((f"A{cols}", f"{well}{1}", f"{well}{cols}"))
+                self._calibration_combo.addItems(
+                    [
+                        "1 Well (A1)",
+                        f"2 Wells (A1,  A{cols})",
+                    ]
+                )
+
+            if from_combo:
+                self._calibration_combo.setCurrentText(from_combo)
+
+        n_tables = self._calibration_combo.currentText()[0]
+        self._show_hide_tables(int(n_tables), well_list)
+
+        if self._calibration_combo.currentText()[0] == "1":
+            wells_to_calibrate = self._calibration_combo.currentText()[8:-1]
+        else:
+            wells_to_calibrate = self._calibration_combo.currentText()[9:-1]
+
+        if self.plate.get("circular"):  # type: ignore
+            text = (
+                f"Calibrate Wells: {wells_to_calibrate}\n"
+                "\n"
+                "Add 3 points on the circonference of the round well "
+                "and click on 'Calibrate Plate'."
+            )
+        else:
+            text = (
+                f"Calibrate Wells: {wells_to_calibrate}\n"
+                "\n"
+                "Add 2 points (opposite vertices) "
+                "or 4 points (1 point per side) "
+                "and click on 'Calibrate Plate'."
+            )
+        self.info_lbl.setText(text)
+
+    def _set_calibrated(self, state: bool) -> None:
+        if state:
+            self.is_calibrated = True
+            self.icon_lbl.setPixmap(
+                icon(MDI6.check_bold, color=(0, 255, 0)).pixmap(QSize(20, 20))
+            )
+            self.cal_lbl.setText("Plate Calibrated!")
+        else:
+            self.is_calibrated = False
+            self.A1_well = ()  # type: ignore
+            self.plate_rotation_matrix = None  # type: ignore
+            self.plate_angle_deg = 0.0
+            self.A1_stage_coords_center = ()
+            self.icon_lbl.setPixmap(
+                icon(MDI6.close_octagon_outline, color="magenta").pixmap(QSize(30, 30))
+            )
+            self.cal_lbl.setText("Plate non Calibrated!")
+
+    def _get_circle_center_(
+        self, a: Tuple[float, float], b: Tuple[float, float], c: Tuple[float, float]
+    ) -> Tuple[float, float]:
+        """Find the center of a round well given 3 edge points."""
+        # eq circle (x - x1)^2 + (y - y1)^2 = r^2
+        # for point a: (x - ax)^2 + (y - ay)^2 = r^2
+        # for point b: = (x - bx)^2 + (y - by)^2 = r^2
+        # for point c: = (x - cx)^2 + (y - cy)^2 = r^2
+
+        x1, y1 = a
+        x2, y2 = b
+        x3, y3 = c
+
+        x, y = symbols("x y")
+
+        eq1 = Eq(
+            (x - round(x1)) ** 2 + (y - round(y1)) ** 2,
+            (x - round(x2)) ** 2 + (y - round(y2)) ** 2,
+        )
+        eq2 = Eq(
+            (x - round(x1)) ** 2 + (y - round(y1)) ** 2,
+            (x - round(x3)) ** 2 + (y - round(y3)) ** 2,
+        )
+
+        dict_center = solve((eq1, eq2), (x, y))
```

Comment:
yeah, I think this can be done with `np.linalg.solve`


github copilot gives me:
```
def circle_from_3_points(
    c1: np.ndarray, c2: np.ndarray, c3: np.ndarray
) -> Tuple[np.ndarray, float]:
    """Return the center and radius of the circle passing through the three points."""
    A = np.array([[c1[0], c1[1], 1], [c2[0], c2[1], 1], [c3[0], c3[1], 1]])
    b = np.array([c1[0] ** 2 + c1[1] ** 2, c2[0] ** 2 + c2[1] ** 2, c3[0] ** 2 + c3[1] ** 2])
    x = np.linalg.solve(A, b)
    center = np.array([x[0], x[1]])
    radius = np.sqrt(x[2] + x[0] ** 2 + x[1] ** 2)
    return center, radius
```

but haven't checked it